sdk/storage/src/table/clients/entity_client.rs

```diff
@@ -1,355 +1,289 @@
 use crate::table::prelude::*;
-use crate::table::requests::*;
-use bytes::Bytes;
+use azure_core::{Context, Error, PipelineContext};
+use bytes::Buf;
 use http::method::Method;
-use http::request::{Builder, Request};
-use std::sync::Arc;
-use url::Url;
+use once_cell::sync::Lazy;
+use serde::de::DeserializeOwned;
+use std::borrow::Cow;
 
-pub trait AsEntityClient<RK: Into<String>> {
-    fn as_entity_client(&self, row_key: RK) -> Result<Arc<EntityClient>, url::ParseError>;
-}
-
-impl<RK: Into<String>> AsEntityClient<RK> for Arc<PartitionKeyClient> {
-    fn as_entity_client(&self, row_key: RK) -> Result<Arc<EntityClient>, url::ParseError> {
-        EntityClient::new(self.clone(), row_key)
-    }
-}
+// we need this since the http::Method does not have the MERGE verb. The unwrap is safe here.
+static MERGE: Lazy<http::Method> = Lazy::new(|| http::Method::from_bytes(b"MERGE").unwrap());
 
-#[derive(Debug, Clone)]
 pub struct EntityClient {
-    partition_key_client: Arc<PartitionKeyClient>,
-    row_key: String,
-    url: Url,
+    table_name: Cow<'static, str>,
```

Comment:
I would prefer a simpler `String`. Table names should be relatively small anyway.